hatice-ecevit-2016400138/assignment1/KthtoLast.java

```diff
@@ -0,0 +1,46 @@
+public class KthtoLast {
+    public static void main(String[] args) {
+        Node<Integer> n1 = new Node<>(3);
+        Node<Integer> n2 = new Node<>(8, n1);
+        Node<Integer> n3 = new Node<>(9, n2);
+        Node<Integer> n4 = new Node<>(7, n3);
+        Node<Integer> n5 = new Node<>(5, n4);
+        Node<Integer> n6 = new Node<>(2, n5);
+        Node<Integer> n7 = new Node<>(0, n6);
+
+        System.out.println(kthToLastElement(n7, 3).getData());
+
+    }
+
+    /**
+     *
+     * @param head:head of the given linked list
+     * @param k the index (from the end of the list) of the element to be returned
+     * @return the node corresponding to the k-th to last element of the list
+     * @returns null if the head of the list is null.
```

Comment:
Instead of `if the head of the list is null` (which is specific to your implementation), you can say `if the list is empty`, which is what a user of the API would be thinking about.